core/tests/protractor_mobile/navigation.js

```diff
@@ -0,0 +1,168 @@
+      waitFor.elementToBeClickable(
+        teachLink, 'Could not click teach link');
+      teachLink.click();
+      waitFor.pageToFullyLoad();
+      expect(browser.getCurrentUrl()).toEqual('http://localhost:9001/teach');
+    });
+
+  it('should navigate to Donate page using the sidebar menu',
+    function() {
+      var navbarButton = element(
+        by.css('.protractor-mobile-test-navbar-button'));
+      waitFor.elementToBeClickable(
+        navbarButton, 'Could not click navbar button');
+      navbarButton.click();
+      var donateLink = element(
+        by.css('.protractor-mobile-test-donate-link'));
+      waitFor.elementToBeClickable(
+        donateLink, 'Could not click donate link');
+      donateLink.click();
+      waitFor.pageToFullyLoad();
+      expect(browser.getCurrentUrl()).toEqual('http://localhost:9001/donate');
+    });
+
+  it('should navigate to Contact page using the sidebar menu',
+    function() {
+      var navbarButton = element(
+        by.css('.protractor-mobile-test-navbar-button'));
+      waitFor.elementToBeClickable(
+        navbarButton, 'Could not click navbar button');
+      navbarButton.click();
+      var contactLink = element(
+        by.css('.protractor-mobile-test-contact-link'));
+      waitFor.elementToBeClickable(
+        contactLink, 'Could not click contact link');
+      contactLink.click();
+      waitFor.pageToFullyLoad();
+      expect(browser.getCurrentUrl()).toEqual('http://localhost:9001/contact');
+    });
+
+  it('should navigate to Library page using the sidebar menu',
+    function() {
+      var navbarButton = element(
+        by.css('.protractor-mobile-test-navbar-button'));
+      waitFor.elementToBeClickable(
+        navbarButton, 'Could not click navbar button');
+      navbarButton.click();
+      var libraryLink = element(by.css('.protractor-mobile-test-library-link'));
+      waitFor.elementToBeClickable(
+        libraryLink, 'Could not click library link');
+      libraryLink.click();
+      waitFor.pageToFullyLoad();
+      expect(browser.getCurrentUrl()).toEqual('http://localhost:9001/library');
+    });
+
+  it('should navigate to Splash page by clicking on the Oppia logo',
+    function() {
+      var oppiaLogo = element(by.css('.protractor-test-oppia-main-logo'));
+      waitFor.elementToBeClickable(
+        oppiaLogo, 'Could not click oppia logo');
+      oppiaLogo.click();
+      waitFor.pageToFullyLoad();
+      expect(browser.getCurrentUrl()).toEqual('http://localhost:9001/splash');
+    });
+
+  it('should navigate to the login page by clicking on the sign in button',
+    function() {
+      var signInButton = element(
+        by.css('.protractor-mobile-test-login'));
+      waitFor.elementToBeClickable(
+        signInButton, 'Could not click sign in button');
+      // For the rationale behind this statement, see
+      // https://github.com/angular/protractor/issues/2643#issuecomment-213257116
+      browser.ignoreSynchronization = true;
```

Comment:
Just to check, do we still need this stuff after all the "wait" refactoring? /cc @hoangviet1993

Reply:
Thanks for mention @seanlip!
Apurv and I also discussed this , somewhat extensively, over Gitter. The last time I did this in the middle of (former) learnerDashboard, it came back to bite hard as the flakiness doubled down on PR builds as time went on.
Flash forward to today, Apurv mentioned that adding Protractor's ignoreSync at that location seems to fix the "Could not find Angular" error.
I responded that is great news because until now, we are under the mercy of Travis not running into that error in daily PR builds randomly and reset the whole test.
This is the needed context of why that statement is there. I have some partial knowledge about it being there given the forewarn from Apurv.

Reply:
Ever since #5139, I have been pretty outspoken about my uncertainty about Protractor's ignoreSync ( synonym for waitForAngularEnabled ). It seems to be a bandage for crossing non-Angular page over into Angular land.

Last time, Sean saved whats left of my sanity after #5139 by telling me to stop refreshing the page and just click the logo. This removed the need for ignoreSync.

This time, again, I am not sure what is the solution as to replace ignoreSync. That ignoreSync line could either save us a lot of time in daily PR builds __or___ turn whichever particular suite using it into flaky town.

Reply:
Let me once state the purpose of this: if we look at the spec closely, we are transitioning from an angular page (Library page) to a non-angular page (Login page) using the "Sign In" button. Protractor, by default, waits for angular to load completely. Since there is no angular on the login page, the test times out saying, "Could not find Angular on this page...". The `browser.ignoreSynchronization = true` asks Protractor not to wait for the Angular page. As soon as this page loads up, we are again setting `browser.ignoreSynchronization = false`, therefore there is no issue of flakiness here.

Reply:
Ah OK. I'm fine with leaving as-is, just wanted to double-check.

@apb7, you have a lot of good context in your last comment, can we add it to the comment in the code here?

Thanks!

Reply:
Added detailed comments explaining the need of the above statements!